File: FE/src/features/permission/services/permissionService.ts
import { apiClient } from '../../../services/apiClient'
import type { PaginatedResult } from '../../../types/api.types'
import type {
  CreatePermissionRequest,
  PermissionResponse,
  UpdatePermissionRequest,
} from '../types/permission.types'

export const permissionService = {
  getAll: (page = 1, size = 10, sort = 'id,asc') => {
    const params = new URLSearchParams({
      page: String(page),
      size: String(size),
      sort,
    })
    return apiClient<PaginatedResult<PermissionResponse>>(`/permissions?${params}`)
  },

  getById: (id: number) => apiClient<PermissionResponse>(`/permissions/${id}`),

  create: (data: CreatePermissionRequest) =>
    apiClient<PermissionResponse>('/permissions', {
      method: 'POST',
      body: JSON.stringify(data),
    }),

  update: (data: UpdatePermissionRequest) =>
    apiClient<PermissionResponse>('/permissions', {
      method: 'PUT',
      body: JSON.stringify(data),
    }),

  delete: (id: number) =>
    apiClient<null>(`/permissions/${id}`, { method: 'DELETE' }),
}

File: BE/src/main/java/Manager_vnd/Manager/feature/permission/PermissionService.java
package Manager_vnd.Manager.feature.permission;

import Manager_vnd.Manager.dto.PaginatedResult;
import Manager_vnd.Manager.feature.permission.dto.CreatePermissionRequest;
import Manager_vnd.Manager.feature.permission.dto.PermissionResponse;
import Manager_vnd.Manager.feature.permission.dto.UpdatePermissionRequest;

public interface PermissionService {

    PaginatedResult<PermissionResponse> getAll(int page, int size, String sort);

    PermissionResponse getById(long id);

    PermissionResponse create(CreatePermissionRequest request);

    PermissionResponse update(UpdatePermissionRequest request);

    void delete(long id);
}
